### ISSUE TITLE
Report button and artifacts giving invalid build specification

### ISSUE THREAD

**Issue by andersoncursino** [YOU54F/cypress-slack-reporter#1037]
I tried to do everything like the README and set the export and everything but it keep me giving some "invalid build specification" with undefined on the URL.

Maybe because the new changes to the URL artifact method? or I'm doing something wrong on export?
My export:
`export CIRCLE_PROJECT_ID=${CI_PROJECT_ID}`

Any help will be much appreciated! Thank you in advance!

Below the result when I try to access:
![image](https://user-images.githubusercontent.com/683768/126681980-3686135f-b7bb-4e90-a67b-258760c0c2a9.png)

**Comment by YOU54F** [comment]
If you type echo $ci_project_id what is your output? Is that defined?

**Comment by YOU54F** [comment]
Also thanks for being super polite. The readme could do with some love. I find it hard to follow 😂

**Comment by andersoncursino** [comment]
Thanks for the super fast answer, when I try to SSH into Circle Ci image this is what I got:
```
root@123:~/app# echo $CIRCLE_PROJECT_ID

root@123:~/app# echo ${CI_PROJECT_ID}

root@123:~/app# echo $CIRCLE_BUILD_URL
https://circleci.com/gh/projectname/repositoryname/36
root@a899931c1492:~/app# 
```

Both variables give empty results: $CIRCLE_PROJECT_ID and ${CI_PROJECT_ID}
But I'm able to get other environment variables.

**Comment by andersoncursino** [comment]
Also, your README is very good, I'm only asking because I don't see on circle ci environment variables the CI_PROJECT_ID or CIRCLE_PROJECT_ID so I don't know if is something that I'm doing wrong or missing here.

**Comment by YOU54F** [comment]
If you got into the artefacts section in circle CI job run and click on a link for the artefact, the project id should be in the url.  There is a link in the readme I think for the circleci page description the opt for how to get it

https://github.com/YOU54F/cypress-slack-reporter#circleci-artifact-notes

You will need to determine it yourself via an api call or clicking on the artefact tab then link as descrived above it wasn’t exposed on the front end site as far as I could tell last time I looked

**Comment by andersoncursino** [comment]
I tried but unfortunately, the API call is returning empty for me all the time, I tried all things and I really don't know how to fix that.

One thing I saw is if I try to access the artifacts only using: `https://circleci.com/api/v1.1/project/github/username/project/build_num/artifacts/0/mochareports/report.html`

I got success in retrieving the artifact, there is any way to change the URL or any option to insert a custom without using the custom script?

Like:
`https://circleci.com/api/v1.1/project/github/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_BUILD_NUM} /artifacts/0/mochareports/report.html`

I really struggling on retrieving the project id and got success on the report artifact

Really appreciate all the patience to help me! Thanks!

**Comment by andersoncursino** [comment]
Ok I was able to get the project ID and export as variable, but I'm still getting the undefined, if I echo on workflow I can see the CIRCLE_PROJECT_ID successful exported
![image](https://user-images.githubusercontent.com/683768/126884782-c842630b-c504-4860-9675-3ccccc813b8f.png)

I really appreciate if you can help me and point the right direction, any other tips or maybe see if I'm still doing something wrong?

thanks in advance!

**Comment by andersoncursino** [comment]
Never mind, totally forgot that circle ci don't pass env var to other steps, worked, thank you for point on me right direction on the first message! thank you so much for this awesome report tool to slack!

**Comment by YOU54F** [comment]
Awesome glad it’s sorted and no worries, glad to be able to share for others ✌🏽
